includes/class-sensei-frontend.php

```diff
@@ -151,6 +151,8 @@ public function enqueue_scripts() {
 			wp_register_script( 'sensei-course-video-youtube-iframe-api', 'https://www.youtube.com/iframe_api', [], 'unversioned', true );
 			Sensei()->assets->register( 'sensei-course-video-blocks-youtube', 'js/frontend/course-video/youtube-extension.js', [ 'sensei-course-video-youtube-iframe-api' ], true );
 
+			Sensei()->assets->register( 'sensei-course-video-blocks-video', 'js/frontend/course-video/video-extension.js', [], true );
```

Comment:
About the scripts registration, they could be all in the same script too, right?

Reply:
Actually, we don't want to load all of them always.
And there is also one more thing. Some of them (YouTube and Vimeo) have dependencies -- and it was really easy to manage them by enqueueing separate scripts. 
Are there other options how to handle that?

Reply:
I think in 2 ways that we could do that.

1) We could enqueue them separately when we enqueue the assets, instead of a dependendency. We could enqueue them based on the [has_block](https://developer.wordpress.org/reference/functions/has_block/), or something like that.
2) I think both are very small, right? We could include both as dependencies, even if they're using only youtube embed, vimeo embed, or video block.

Reply:
You're right. These scripts seem tiny. So I want to go further with the second option.